### PR TITLE
[GH-967] fix overflow URL property in cell

### PR DESCRIPTION
**Pull request by jufab**
#### Summary

This PR add some css properties to fix overflow in URL property cell
And add a unit test for this component

#### Ticket Link

https://github.com/mattermost/focalboard/issues/967

### CODE REVIEW

File: webapp/src/components/properties/link/link.test.tsx
Comment (kamre) [inline webapp/src/components/properties/link/link.test.tsx:24]: How is this test related to the changes in `link.scss`?

Comment (jufab) [reply webapp/src/components/properties/link/link.test.tsx:24]: You right.
I used an existing css class at the beginning...
I'm going to fix that 😊

Comment (jufab) [reply webapp/src/components/properties/link/link.test.tsx:24]: That's not possible. 😢 
Jest configured with a mock on scss

```
"moduleNameMapper": {
    "\\.(scss|css)$": "<rootDir>/__mocks__/styleMock.js"
},
```
And styleMock.js is empty export

```
module.exports = {};
```

And I don't know how to do....Any idea?

Comment (kamre) [reply webapp/src/components/properties/link/link.test.tsx:24]: I don't think there is an easy way to check css style inside a unit test.

Comment (jufab) [reply webapp/src/components/properties/link/link.test.tsx:24]: Maybe an other issue  to resolve this problem of css in unit tests and do not block the PR for #967?

Comment (kamre) [reply webapp/src/components/properties/link/link.test.tsx:24]: This is definitely not blocking. And even such unit test has value. Just need approval from core contributors (look at the list of reviewers).

**Comment by hahmadia** [comment]
Hi @jufab 

Thank you for your contribution. I'm going to add my colleague @asaadmahmood to review your SCSS changes since he's quite the expert to make sure these are in fact the changes we want.

**Review by wiggin77 — APPROVED** [review]
LGTM :+1:   The unit test is not really related to the CSS change but looks fine.

**Review by asaadmahmood — CHANGES_REQUESTED** [review]
The hover behaviour looks weird, as it actually shifts the whole rows down. The icon should instead be positioned on the top right of the cell.

Screenshot 2021-08-18 at 5 06 47 PM

**Comment by kamre** [comment]
@asaadmahmood this behaviour with the link icon appearing below the input is reproducible on `main` currently.

Do you propose to have something like this:
LinkHover
?

**Comment by asaadmahmood** [comment]
@kamre Yup, something like that makes sense.

File: webapp/src/components/properties/link/link.scss
Comment (kamre) [inline webapp/src/components/properties/link/link.scss:4]: By the way why `width: inherit;` is needed here?

Comment (asaadmahmood) [reply webapp/src/components/properties/link/link.scss:4]: Shouldn't be.

**Comment by kamre** [comment]
@jufab @asaadmahmood propose to try this `link.scss` variant:
```scss
.focalboard-body .property-link {
    &.url {
        display: flex;
        overflow: hidden;
    }

    a {
        padding: 0 8px; // increases clickable area for better UX
        align-self: stretch;
        display: none;
    }

    a:hover {
        background:  unset;
    }

    &:hover {
        a {
            display: flex;
            align-items: center;
        }
    }
}
```
- `flex` for `.url` allows to keep `input` and `a` on the same line and also `input` can shrink if needed.
- Vertical padding for `a` is set to `0` so that it doesn't try to increase the table row height when `a` appears on hover.

**Comment by jufab** [comment]
@kamre I prefer yours but without padding-right on `url`

with padding-right on hover : 
link-hover
without padding-right on hover :
link-hover-without-padding

I'm going to push the last one.

**Comment by asaadmahmood** [comment]
@jufab If I expand the width of the column, the url thing doesn't occupy the whole space, also the link icon should appear when you hover over the cell, currently it only appears when I hover over the input.

Screenshot 2021-08-19 at 5 59 01 PM

**Review by asaadmahmood — CHANGES_REQUESTED** [review]
See my comment above.

**Comment by jufab** [comment]
> @jufab If I expand the width of the column, the url thing doesn't occupy the whole space, also the link icon should appear when you hover over the cell, currently it only appears when I hover over the input.
> 
> Screenshot 2021-08-19 at 5 59 01 PM

@asaadmahmood I fix the url to occupy the whole space with a "width: inherit". 

link-hover

But for hover, it's a "cell" problem with a 8px padding

**Comment by asaadmahmood** [comment]
@jufab That's fine, I can have a look at it later.

**Review by asaadmahmood — APPROVED** [review]

**Review by hahmadia — APPROVED** [review]

**Comment by hahmadia** [comment]
Hi! Could please fix the broken snapshot tests so that we can merge this in? Thanks

**Comment by jufab** [comment]
Hi @hahmadia,
it's done!

**Pull request merged**
